### ISSUE TITLE
Using spaces in search query for separate keywords/substrings

### ISSUE THREAD

**Issue by hi2u** [kbysiec/vscode-search-everywhere#5]
Let's say I have a function called `oneTwoThree()` somewhere in my project, but I can't remember the exact name of it (or the order of words)... I would normally search for it with a space between keywords, e.g. search queries like:

* one three
* three one
* on thr
* tw on

But it seems that any query that contains a space gives me zero results.  I think the 2nd+ words only filter on the filepath? (even when I'm searching for symbol names)

Although if I remove the space, the results work.  Would it be possible for the search to work even if you put spaces in between the keywords?  

Given that I seem to need to join the words together, I've noticed while I'll get the correct result for query `onethree`, but I won't for `onethreetwo` - so if I know the three words that exist in a symbol name (but not the exact order), I can't search for it given that I can't seem to use spaces in the query to separate 2+ separate substrings.

**Comment by kbysiec** [comment]
Unfortunately, the current quick pick api doesn't allow for taking over the filtering of items list in easy way. I would even say there is not any way, only workaround.

During last few days I have tested the possibilities and noticed that:
* Quick pick highlighting of current text in search results is not possible to overwrite at all. I have ping the feature request status in vscode repository and received an information it is on the list. Nothing more so there isn't any date we can expect it.
* I have implemented the filtering using fusejs, but it is far away from ideal solution, using threshold 0.5 gives items which shouldn't be listed but 0.4 (closer to exact) not include the ones which should be.

To sum up, currently it is not possible taking into consideration available possibilities given by vscode api.

I am setting the issue as on hold till the time when api allows us to implement the feature.

Regards,
Kamil
